### ISSUE TITLE
Why is the documentation so bad?

### ISSUE THREAD

**Issue by kasajian** [dotnet/corefx#17935]
It's only a problem with Microsoft components.

I see a component in NuGet whose name looks interesting.   I click on Project Site and I am immediately taken to the site for the project that produces the component, which explains what it is, etc.

But not with Microsoft components.   Here's a simple example.   I see a component in NuGet called "System.Threading.Tasks.Extensions".   I click on Project Site.  It takes me to a generic docs site, the main page that has nothing to do with System.Threading.Tasks.Extensions.   If I then search for System.Threading.Tasks.Extensions, I got lots of hits, none of which are for System.Threading.Tasks.Extensions, but System.Threading.Tasks instead.

I do the search again, this time putting the component name in quotes, prepending a plus sign to indicate that I don't want to see any pages unless the exact name appears:
+"System.Threading.Tasks.Extensions"

Now I am given a page that is about another component, value.types, that says I need System.Threading.Tasks.Extensions in order to use it.

Googling or Binging "System.Threading.Tasks.Extensions" doesn't do anything either.

Now, wouldn't it be better if System.Threading.Tasks.Extensions had a documentation page somewhere similar to MSDN, and if one does not exist, actually indicate it does not exist.    If there's not docs, telling the user there's not docs is actually quite valuable.  The user can stop looking for docs, and may be go elsewhere such as stackoverflow, forums or decide to look through the source.

There's a reason why there's a "project site" link on nuget.org.  It's supposed to take you to the specific page for that component.   It would be useful if that's what it did rather than sending you to an entire Docs site where you have to now search for the phrase.

**Comment by danmosemsft** [comment]
@terrajobst

**Comment by karelz** [comment]
cc @mairaw if we have any pages we can link to
cc @leecow @ericstj @weshaggard - where do we fill the docs? During publishing I assume. And we can change it for existing packages (@ericstj pointed me to the instructions recently AFAIK)

**Comment by mairaw** [comment]
I'm not aware of any pages we can link to that already exist. @rpetrusha do you know?

**Comment by svick** [comment]
I think the main problem here is that `System.Threading.Tasks.Extensions` is a *package*, but [the API reference](https://docs.microsoft.com/en-us/dotnet/core/api/) is organized by *namespace*. It would be nice if each package had its own page in the documentation, where the Project Link on NuGet could lead.

Another issue is that the API reference has no documentation for `ValueTask` (currently the only reason why that package exists).

As a short term fix, https://docs.microsoft.com/dotnet/articles/csharp/whats-new/csharp-7#generalized-async-return-types might be the best URL to link from that package. But all the other packages have the same issue too (just usually less pronounced, because their names are clearer or correspond to namespaces).

**Comment by krwq** [comment]
Should we perhaps simply add a README page for each library (under `src/<library>/README.md`) and add few samples per project + links to MSDN?

**Comment by kasajian** [comment]
@krwq I think anything is better than what we have now, and I really do think your proposal is good.  I think that if you even just point to a wiki page, then you'll have someone in the community fill in whatever information they happen to have.

**Comment by karelz** [comment]
I like @krwq's idea -- we need an easy-to-find place (with stable URL we can link from NuGet packages) that folks are **not afraid** to contribute to (both internal and external contributors). I don't think that CoreFX repo is the right place though -- @mairaw can you please guide us where to set up such docs? Would it be best in docs repo? Could we get some help from docs team with vetting the write ups, etc.?

BTW: We might still want to create README.md per library (as @krwq suggests) and put there link to this 'blessed' place where the docs live.

**Comment by kasajian** [comment]
Maybe there is a single service that can be used such that, based on the query in the URL, it will go to a page for that particular NuGet component.   Or, maybe, the service can use the originating web-page to determine which component this is for.   Anyway, once there, there can be a page that's both official docs, and community docs.   That means that, from the get-go, every component will at a minimum have a community-page.

For instance, someone from the .NET Foundation / Core, etc.  publishes to NuGet.   For the Project Site, they specify some specific URL.   Once published, if a user clicks on that link, they're sent to a doc page.  The doc page may not have anything in it yet.  but there's a community page link at the bottom (which also may not have anything in it), but people can at least start posting samples to help others.

**Comment by svick** [comment]
@kasajian I don't see any reason to have separate official and community docs. Whatever decision is made about where package documentation should be, it should be both official and accepting community contributions, just like almost everything related to .Net Core, including documentation, already is.

**Comment by krwq** [comment]
@svick fair point, I find README pages easiest to find and contribute. Usually any docs which are not close to a source code are quite outdated mostly because it is not obvious where do you change it. Whatever we choose it should be obvious how to contribute
